`test_multiple_disks::test_concurrent_alter_move[concurrently_altering_replicated_mt-ReplicatedMergeTree('/clickhouse/concurrently_altering_replicated_mt', '1')]` fails

```
______ test_concurrent_alter_move[concurrently_altering_replicated_mt-ReplicatedMergeTree('/clickhouse/concurrently_altering_replicated_mt', '1')] ______

start_cluster = <helpers.cluster.ClickHouseCluster instance at 0x7f9ca6aa5290>, name = 'concurrently_altering_replicated_mt'
engine = "ReplicatedMergeTree('/clickhouse/concurrently_altering_replicated_mt', '1')"

    @pytest.mark.parametrize("name,engine", [
        ("concurrently_altering_mt","MergeTree()"),
        ("concurrently_altering_replicated_mt","ReplicatedMergeTree('/clickhouse/concurrently_altering_replicated_mt', '1')",),
    ])
    def test_concurrent_alter_move(start_cluster, name, engine):
        try:
            node1.query("""
                CREATE TABLE {name} (
                    EventDate Date,
                    number UInt64
                ) ENGINE = {engine}
                ORDER BY tuple()
                PARTITION BY toYYYYMM(EventDate)
                SETTINGS storage_policy='jbods_with_external'
            """.format(name=name, engine=engine))
    
            def insert(num):
                for i in range(num):
                    day = random.randint(11, 30)
                    value = random.randint(1, 1000000)
                    month = '0' + str(random.choice([3, 4]))
                    node1.query("INSERT INTO {} VALUES(toDate('2019-{m}-{d}'), {v})".format(name, m=month, d=day, v=value))
    
            def alter_move(num):
                for i in range(num):
                    produce_alter_move(node1, name)
    
            def alter_update(num):
                for i in range(num):
                    node1.query("ALTER TABLE {} UPDATE number = number + 1 WHERE 1".format(name))
    
            def optimize_table(num):
                for i in range(num):
                    node1.query("OPTIMIZE TABLE {} FINAL".format(name))
    
            p = Pool(15)
            tasks = []
            for i in range(5):
                tasks.append(p.apply_async(insert, (100,)))
                tasks.append(p.apply_async(alter_move, (100,)))
                tasks.append(p.apply_async(alter_update, (100,)))
                tasks.append(p.apply_async(optimize_table, (100,)))
    
            for task in tasks:
                task.get(timeout=60)
    
            assert node1.query("SELECT 1") == "1\n"
>           assert node1.query("SELECT COUNT() FROM {}".format(name)) == "500\n"
E           AssertionError: assert '499\n' == '500\n'
E             - 499
E             + 500
```
[clickhouse-server.log](https://github.com/ClickHouse/ClickHouse/files/4051485/clickhouse-server.log)

--- a/dbms/tests/integration/test_multiple_disks/test.py
+++ b/dbms/tests/integration/test_multiple_disks/test.py
@@ -686,10 +686,12 @@ def test_concurrent_alter_move(start_cluster, name, engine):
             SETTINGS storage_policy='jbods_with_external'
         """.format(name=name, engine=engine))
 
+        values = list({ random.randint(1, 1000000) for _ in range(0, 1000) })
+
         def insert(num):
             for i in range(num):
                 day = random.randint(11, 30)
-                value = random.randint(1, 1000000)
+                value = values.pop()
                 month = '0' + str(random.choice([3, 4]))
                 node1.query("INSERT INTO {} VALUES(toDate('2019-{m}-{d}'), {v})".format(name, m=month, d=day, v=value))
 

